src/st/no0/e_ctulhu.c

```diff
@@ -581,8 +583,217 @@ void EntityCtulhuFireball(Entity* self) {
+                prim->clut = 0x234;
+                PGREY(prim, 0) = PGREY(prim, 1) = PGREY(prim, 2) =
+                    PGREY(prim, 3) = 96;
+                prim->x0 = prim->x2 = self->posX.i.hi + 16;
+                prim->x1 = prim->x3 = self->posX.i.hi - 16;
+                prim->y0 = prim->y1 = self->posY.i.hi - 24;
+                prim->y2 = prim->y3 = self->posY.i.hi + 48;
+                prim->priority = self->zPriority;
+                prim->drawMode = DRAW_TPAGE2 | DRAW_TPAGE | DRAW_COLORS |
+                                 DRAW_UNK02 | DRAW_TRANSP;
+            }
+        } else {
+            self->ext.ctulhu.timer -= 1;
+        }
+
+        if (self->flags & FLAG_DEAD) {
+            self->drawFlags |= FLAG_DRAW_UNK8;
+            self->unk6C = 128;
+            self->hitboxState = 0;
+            self->step++;
+        }
+        break;
+    case 2:
+        MoveEntity();
+        self->unk6C -= 8;
+        self->rotY += 24;
+        self->velocityX -= self->velocityX / 8;
+
+        if (g_Timer & 1) {
+            self->animCurFrame = 0;
+        } else {
+            self->animCurFrame = 44;
+        }
+
+        if (!self->unk6C) {
+            DestroyEntity(self);
+            return;
+        }
+        break;
+    }
+
+    prim = self->ext.prim;
+    while (prim != NULL) {
+        if (!(prim->drawMode & DRAW_HIDE)) {
+            if (prim->p3 == 2) {
+                PrimDecreaseBrightness(prim, 5);
+
+                if (g_Timer & 1) {
+                    prim->u0 = prim->u2 = 4;
+                    prim->u1 = prim->u3 = 28;
+                } else {
+                    prim->u0 = prim->u2 = 28;
+                    prim->u1 = prim->u3 = 4;
+                }
+
+                prim->y0 = (prim->y1 += 3);
+                prim->v2 = (prim->v3 -= 1);
+
+                prim->x2 -= (F(self->velocityX).i.hi) >> 1;
+                prim->x3 += (F(self->velocityX).i.hi) >> 1;
+                if (prim->x0 != prim->x2) {
+                    prim->x0 = prim->x2;
+                    prim->x1 = prim->x3;
+                } else {
+                    prim->x0 = prim->x2 + 4;
+                    prim->x1 = prim->x3 + 4;
+                }
+
+                if (prim->y0 > prim->y2 || prim->v2 < prim->v0) {
+                    prim->p3 = 0;
+                    prim->drawMode = DRAW_HIDE;
+                }
+            }
+
+            if (prim->p3 == 4) {
+                PrimDecreaseBrightness(prim, 4);
+                if (prim->y2 - prim->y0 > 32) {
+                    prim->y0 = (prim->y1 += 2);
+                }
+                if (!prim->p2) {
+                    prim->p1++;
+                    if (prim->p1 > 13) {
+                        prim->p3 = 0;
+                        prim->drawMode = DRAW_HIDE;
+#ifdef VERSION_PSP
+                        // Looks to be a bug fix on PSP
+                        // PSX can index outside size of D_us_801C11B0
+                        goto primNext;
```

Comment:
There's a few ways to accomplish a match here, but I couldn't find a way without PSP defs.

One alternative is doing
```c
if (prim->p1 > 13) {
...
}
#ifdef VERSION_PSP
else {
#endif
...
#ifdef VERSION_PSP
}
#endif
```

which also matches, I can change to that if it feels more intuitive than a goto.

I think this is a bug fix on PSP since `D_us_801C11B0` has 14 elements in the data, and on PSX this code can index past the end of the array after the prim gets hidden

Reply:
The non-goto version is more clear to me so that would be my vote

Reply:
Updated